### ISSUE TITLE
a11y: keyboard navigation is broken for filtered survey menus

### ISSUE THREAD

**Issue by eatyourgreens** [zooniverse/front-end-monorepo#7336]
## Package
- lib-classifier

## Describe the bug
Here, I'm using the keyboard to navigate a survey task after filtering the list of choices. I can enter the grouped list, but can't then navigate the choices with the cursor keys.

https://github.com/user-attachments/assets/aaff414b-c915-48d8-beb5-d7c1208cd779

## To Reproduce
https://www.zooniverse.org/projects/drkellie/wildlife-recovery-count/classify/workflow/31257

Using only the keyboard, choose a filter, make a choice, and submit your classification.

## Expected behavior
I should be able to classify using only the keyboard.

## Device information
Firefox 150 on MacOS Sequoia, with keyboard navigation enabled in accessibility settings.

## Additional context
[WCAG SC 2.1.1 Keyboard](https://www.w3.org/WAI/WCAG22/Understanding/keyboard.html) is a level A requirement ie. failure means people with mobility impairments are excluded from participating in the project.

**Comment by eatyourgreens** [comment]
Classifying works fine from the keyboard if I don't choose a filter. This bug is specific to filtered lists.

**Comment by eatyourgreens** [comment]
I just learnt about the `roving-tabindex` web component, which handles focus management for menus and grids. 

https://npmx.dev/package/roving-tabindex

Blink also announced intent to ship for the `focusgroup` attribute. 

https://groups.google.com/a/chromium.org/g/blink-dev/c/53PP0LgddCQ/m/Be_6nT9VBgAJ
